com.unity.render-pipelines.core/Runtime/Volume/VolumeParameter.cs

```diff
@@ -1641,14 +1643,227 @@ internal override void Interp(VolumeParameter from, VolumeParameter to, float t)
+
+                        if (lhsKey.time == rhsKey.time)
+                        {
+                            lhsKeyCurr++;
+                            rhsKeyCurr++;
+                        }
+                        else if (lhsKey.time < rhsKey.time)
+                        {
+                            // in this case:
+                            //     rhsKey[curr-1].time <= lhsKey.time <= rhsKey[curr].time
+                            // so interpolate rhsKey at the lhsKey.time.
+                            rhsKey = EvalKeyAtTime(rhsCurve.keys, rhsKeyCurr - 1, rhsKeyCurr, startTime, endTime, lhsKey.time);
+                            lhsKeyCurr++;
+                        }
+                        else
+                        {
+                            // only case left is (lhsKey.time > rhsKey.time)
+                            Assert.IsTrue(lhsKey.time > rhsKey.time);
+
+                            // this is the reverse of the lhs key case
+                            //     lhsKey[curr-1].time <= rhsKey.time <= lhsKey[curr].time
+                            // so interpolate lhsKey at the rhsKey.time.
+                            lhsKey = EvalKeyAtTime(lhsCurve.keys, lhsKeyCurr - 1, lhsKeyCurr, startTime, endTime, rhsKey.time);
+                            rhsKeyCurr++;
+                        }
+                    }
+                    else if (lhsValid)
+                    {
+                        // we are still processing lhsKeys, but we are out of rhsKeys, so increment lhs and evaluate rhs
+                        lhsKey = lhsCurve.keys[lhsKeyCurr];
+
+                        // rhs will be evaluated between the last rhs key and the extrapolated rhs key at the end time
+                        rhsKey = EvalKeyAtTime(rhsCurve.keys, rhsKeyCurr - 1, rhsKeyCurr, startTime, endTime, lhsKey.time);
+
+                        lhsKeyCurr++;
+                    }
+                    else
+                    {
+                        // either lhsValid is True, rhsValid is True, or they are both True. So to miss the first two cases,
+                        // right here rhsValid must be true.
+                        Assert.IsTrue(rhsValid);
+
+                        // we still have rhsKeys to lerp, but we are out of lhsKeys, to increment rhs and evaluate lhs
+                        rhsKey = rhsCurve.keys[rhsKeyCurr];
+
+                        // lhs will be evaluated between the last lhs key and the extrapolated lhs key at the end time
+                        lhsKey = EvalKeyAtTime(lhsCurve.keys, lhsKeyCurr - 1, lhsKeyCurr, startTime, endTime, rhsKey.time);
+
+                        rhsKeyCurr++;
+                    }
+
+                    Keyframe dstKey = LerpKeyframeDirect(lhsKey, rhsKey, t);
+                    dstKeys[currNumKeys] = dstKey;
+                    currNumKeys++;
+                }
+
+                m_Value = new AnimationCurve();
```

Comment:
This still create GC. Try to find a way to cache it.

Reply:
PR has been updated. Please take a look. Major changes:
1. Moved to a separate KeyframeUtility.cs file.
2. Added tests.
3. To solve the GC issue, the volumes are always blended with the temporary volume structure on the left side (it's already hardcoded to 32 volumes). So instead of lerping lhs and rhs and returning the result, the new version lerps them, and re-uses the lhs AnimationCurve class.

Reply:
Also, I think Felipe is right about the test location. Moved it to under core (VolumeComponentTests.cs) as that seems like a more logical place for the test to live.